staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -907,6 +913,7 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericapifilters.WithRequestDeadline(handler, c.AuditBackend, c.AuditPolicyRuleEvaluator,
 		c.LongRunningFunc, c.Serializer, c.RequestTimeout)
 	handler = genericfilters.WithWaitGroup(handler, c.LongRunningFunc, c.NonLongRunningRequestWaitGroup)
+	handler = genericfilters.WithWatchTerminationDuringShutdown(handler, c.lifecycleSignals, c.WatchRequestWaitGroup)
```

Comment:
This is now always enabled by default, which I don't think is the right thing to do.

Instead, I would expect exposing grace-period as an option (flag) with default value of 0 (backward compatibility) and in case it is set to 0 treat it as disabled.

Reply:
we made it configurable